files/class-vip-filesystem-stream-wrapper.php

```diff
@@ -239,12 +239,14 @@ public function stream_open( $path, $mode, $options, &$opened_path ) {
 	public function stream_close() {
 		$this->debug( sprintf( 'stream_close => %s + %s', $this->path, $this->uri ) );
 
+		$result = true;
+
 		// Don't attempt to flush new file when in read mode
 		if ( $this->should_flush_empty && 'r' !== $this->mode ) {
-			$this->stream_flush();
+			$result = $this->stream_flush();
 		}
 
-		return $this->close_handler( $this->file );
+		return $this->close_handler( $this->file ) && $result;
```

Comment:
The order matters - if we check `$result` first and it is `false`, PHP won't run `close_handler()`.